bson/example_test.go

```diff
@@ -0,0 +1,137 @@
+package bson_test
+
+import (
+	"fmt"
+	"time"
+
+	"go.mongodb.org/mongo-driver/bson"
+)
+
+// This example uses Raw to skip parsing a nested document in a BSON message.
+func ExampleRaw_unmarshal() {
+	b, err := bson.Marshal(bson.M{
+		"Word":     "beach",
+		"Synonyms": bson.A{"coast", "shore", "waterfront"},
+	})
+	if err != nil {
+		panic(err)
+	}
+
+	var res struct {
+		Word     string
+		Synonyms bson.Raw // Don't parse the whole list, we just want to count the elements.
+	}
+
+	err = bson.Unmarshal(b, &res)
+	if err != nil {
+		panic(err)
+	}
+	elems, err := res.Synonyms.Elements()
+	if err != nil {
+		panic(err)
+	}
+	fmt.Printf("%s, synonyms count: %d\n", res.Word, len(elems))
+
+	// Output: beach, synonyms count: 3
+}
+
+// This example uses Raw to add a precomputed BSON document during marshal.
+func ExampleRaw_marshal() {
+	precomputed, err := bson.Marshal(bson.M{"Precomputed": true})
+	if err != nil {
+		panic(err)
+	}
+
+	msg := struct {
+		Message  string
+		Metadata bson.Raw
+	}{
+		Message:  "Hello World!",
+		Metadata: precomputed,
+	}
+
+	b, err := bson.Marshal(msg)
+	if err != nil {
+		panic(err)
+	}
+	// Convert to a bson.Raw to print the BSON document as Extended JSON.
```

Comment:
Optional: IMO this reads more clearly. If you decide to go with this language, there is another instance of this sentence in the "ExampleRawValue_marshal()" function.

```suggestion
	// Print the Extended JSON by converting BSON to bson.Raw.
```